src/index.tsx

```diff
@@ -4,24 +4,84 @@ import "core-js/es6/set";
 
 import React from "react";
 import { render } from "react-dom";
-import { Provider } from "react-redux";
+import { Provider, useDispatch } from "react-redux";
 import { Layout } from "antd";
-import { BrowserRouter, Switch, Route } from "react-router-dom";
+import { BrowserRouter, Switch, Route, useLocation } from "react-router-dom";
 
 import { APP_ID } from "./constants";
 
 import { createReduxStore } from "./state";
-import routes from "./routes";
+import routes, { VIEWER_PATHNAME } from "./routes";
 import ScrollToTop from "./components/ScrollToTop";
 import AppHeader from "./containers/AppHeader";
 
 const { Header } = Layout;
 
 import "./style.css";
+import { setIsPlaying } from "./state/selection/actions";
+import { clearSimulariumFile } from "./state/metadata/actions";
+import {
+    getSimulariumController,
+    getSimulariumFile,
+} from "./state/metadata/selectors";
 
 export const store = createReduxStore();
+interface LocationWithState extends Location {
+    state: {
+        localFile?: boolean;
+    };
+}
+function useLocationChangeToStopPlay() {
```

Comment:
Maybe this function should be named something more like `cleanUpViewerState`? Since stopping play is only done in one possible scenario (leaving viewer), no?